### PR TITLE
vmauth/vmagent:implement configuration initiation for custom config reloader

### PR DESCRIPTION
**Pull request by Haleygo**
implement https://github.com/VictoriaMetrics/operator/issues/619
1. add one-shot mode for config-reloader
2. use one-shot mode in initContainer if `VM_CUSTOMCONFIGRELOADERIMAGE` enabled

### CODE REVIEW

File: internal/config-reloader/main.go [hunk 1]
```diff
@@ -54,6 +55,12 @@ func main() {
 	}
 
 	configWatcher.startWatch(ctx, updatesChan)
+	if *mode == "one-shot" {
+		logger.Infof("config-reloader one-shot mode completed, exit now")
```
Comment (Haleygo) [inline internal/config-reloader/main.go]: If the process[download,parse,write] failed in one-shot mode, should we exit it with code nonzero?

File: internal/config-reloader/main.go [hunk 2]
```diff
@@ -25,6 +25,7 @@ import (
 )
 
 var (
+	mode                   = flag.String("mode", "", "default is normal. If using one-shot mode, will read config and write to config-envsubst-file once")
```
Comment (Haleygo) [inline internal/config-reloader/main.go]: Not sure about the mode name too :)

**Comment by Haleygo** [comment]
@f41gh7 
So the new mode only works in new version of config-reloader, should I add some version control?

File: internal/config-reloader/main.go (hunk 2 above)
Comment (f41gh7) [reply internal/config-reloader/main.go]: I think, boolean flag should be enough  `only-init-config`.

File: internal/config-reloader/main.go (hunk 1 above)
Comment (f41gh7) [reply internal/config-reloader/main.go]: Yep, it should exit with non zero code.

File: internal/config-reloader/main.go [hunk 3]
```diff
@@ -236,7 +245,7 @@ func writeNewContent(data []byte) error {
 		}
 	}
 	tmpDst := *configFileDst + ".tmp"
-	if err := os.WriteFile(tmpDst, data, 0644); err != nil {
+	if err := os.WriteFile(tmpDst, data, 0o644); err != nil {
```
Comment (f41gh7) [inline internal/config-reloader/main.go]: is it a typo?

**Review by f41gh7 — COMMENTED** [review]
Looks good!

**Comment by f41gh7** [comment]
> @f41gh7
> So the new mode only works in new version of config-reloader, should I add some version check in operator or modify the default CustomConfigReloaderImage?

 Yes, it makes sense. I think check for v0.35.0 release is needed. 

 If version cannot be parsed (user defined own docker image with incorrect version name), assume that config-reloader supports new mode.

File: internal/config-reloader/main.go (hunk 3 above)
Comment (Haleygo) [reply internal/config-reloader/main.go]: Sorry, my format tool did that.

**Review by f41gh7 — APPROVED** [review]
LGTM

**Pull request merged**